File: MyTech.Api/Requests/UpdateProductRequest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace MyTech.Api.Requests
{
    public class UpdateProductRequest
    {
        public string newProductText { get; set; }
        public string newSelectedProvider { get; set; }
        public double newProductPrice { get; set; }
        public int newProductStock { get; set; }
    }
}

File: my-tech/src/requests/UpdateProductRequest.js
import axios from 'axios';
import ApiConfig from '../ApiConfig';
import Product from '../representations/Product';

export default class UpdateProductRequest {
    constructor(product, newProductText, newSelectedProvider, newProductPrice, newProductStock) {
        this.product = product;
        this.newProductText = newProductText;
        this.newSelectedProvider =newSelectedProvider;
        this.newProductPrice = newProductPrice;
        this.newProductStock = newProductStock;
    }

    getRequestUrl() {
        const productId = this.product.id;
        return ApiConfig.endpoints.products.updateProduct(productId);
    }

    async send() {
        const requestUrl = this.getRequestUrl();
        const response = await axios.put(requestUrl, {
            newProductText: this.newProductText,
            newSelectedProvider: this.newSelectedProvider,
            newProductPrice: this.newProductPrice,
            newProductStock: this.newProductStock
        });

        return Product.fromApiResponseProduct(response.data);
    }
}